test/util/router.js

```diff
@@ -74,7 +74,7 @@ export const mockRouter = (location = undefined) => (container) => {
       app.component('RouterView', RouterViewStub);
     }
   };
-  for (const prop of ['getRoutes', 'hasRoute', 'resolve'])
+  for (const prop of ['getRoutes', 'hasRoute', 'resolve', 'push'])
```

Comment:
added to test `redirects on submissionsuccess message with return_url`, is there a better way to test that?

Reply:
I have two ideas:

1. Usually if `router.push()` needs to be called, that's a sign that a real router is needed and not `mockRouter()`. `mockRouter()` is just supposed to do basic, read-only things on a single page. So usually that means using `load()`.
2. Maybe it'd be fine to use `mockRouter()` in this case though? The test isn't actually trying to redirect and render a new page, it just wants to see how `router.push()` is called. Maybe we could set `mock.push` to a Sinon fake that we could make assertions about in the test. I'm thinking of:

```js
mock.push = sinon.fake();
```

Then in the test:

```js
router.push.calledWith('/projects/1').should.be.true;
```

Reply:
Now that the test is using `sinon.fake()`, does this file need to change at all? Could we remove `push` from the list?

As a related thought, I'm thinking that it'd probably also be OK for this file to always set `mock.push = sinon.fake()`, if that's something you think would be helpful. Then any future test that needs to make assertions about `push()` would be able to do so easily. We could also wait to make that change later if/when it comes up again.